### PR TITLE
log-out endpoint

### LINKED ISSUE #145: Support Permanent Logging-out
The front-end is not allowed to modify it's own cookies, so we need a back-end endpoint for revoking auth keys.

### PR DESCRIPTION
**Pull request by YRMYJASKA**
Add the logging-out endpoint. Fixed some minor things on the way

Closes #145

### CODE REVIEW

File: internal/webapi/server.go [hunk 1]
```diff
@@ -126,6 +129,15 @@ func (a *Api) Authenticate(auth authentication.Authenticator[APIAuthCredientals]
 	return &femto.EmptyBodyResponse{Cookies: cookies, Status: http.StatusAccepted}, nil
 }
 
+func (a *Api) LogOut(auth authentication.Authenticator[APIAuthCredientals], r *http.Request, l *slog.Logger) (*femto.EmptyBodyResponse, error) {
+	token, err := r.Cookie(authentication.TokenCookieName)
+	if err != nil {
+		return nil, err
+	}
+	auth.RevokeToken(token.Value)
+	return femto.Ok(types.Unit{})
```
Comment (aDotInTheVoid) [inline internal/webapi/server.go:138]: Should this also unset the auth cookie?

File: internal/webapi/server_test.go [hunk 2]
```diff
@@ -13,6 +13,7 @@ import (
 	"github.com/gpuctl/gpuctl/internal/database"
 	"github.com/gpuctl/gpuctl/internal/uplink"
 	"github.com/gpuctl/gpuctl/internal/webapi"
+	"github.com/stretchr/testify/assert"
```
Comment (aDotInTheVoid) [inline internal/webapi/server_test.go:16]: NIT: I'd like to keep 3rd party imports in a seperate group to 1st party (and std)

File: internal/webapi/server_test.go [hunk 3]
```diff
@@ -33,15 +34,52 @@ func TestAuthenticate(t *testing.T) {
...
+	assert.NoError(t, err, "No error in creating auth token")
+
+	// Make new response to revoke the token
+	revokeRequest := httptest.NewRequest(http.MethodGet, "/api/admin/logout", nil)
+	revokeRequest.AddCookie(&http.Cookie{Name: authentication.TokenCookieName, Value: token})
+
+	response, err := api.LogOut(&auth, revokeRequest, mockLogger)
+	assert.NoError(t, err, "No error in logging-out")
+	assert.Equal(t, http.StatusOK, response.Status)
+
```
Comment (aDotInTheVoid) [inline internal/webapi/server_test.go:75]: Maybe worth adding a check that the token no-longer works here.

File: internal/webapi/server_test.go [hunk 4]
```diff
@@ -33,15 +34,52 @@ func TestAuthenticate(t *testing.T) {
...
-	if len(response.Cookies) == 0 {
-		t.Error("Expected an authentication token cookie")
+	found := false
+	for _, cookie := range response.Cookies {
+		if cookie.Name == authentication.TokenCookieName {
+			found = true
+			break
+		}
 	}
+	assert.True(t, found, "Cookies contain auth cookie")
```
Comment (aDotInTheVoid) [inline internal/webapi/server_test.go:44]: Nice catch!

File: internal/webapi/server.go [hunk 5]
```diff
@@ -48,18 +48,21 @@ func NewServer(db database.Database, auth authentication.Authenticator[APIAuthCr
...
+	})
 
 	// Authenticated API endpoints
-	femto.OnPost(mux, "/api/admin/add_workstation", api.addMachine)
-	femto.OnPost(mux, "/api/admin/stats/modify", api.modifyMachineInfo)
-	femto.OnPost(mux, "/api/admin/rm_workstation", api.removeMachine)
-	femto.OnGet(mux, "/api/admin/confirm", func(r *http.Request, l *slog.Logger) (*femto.Response[UsernameReminder], error) {
-		return api.confirmAdmin(auth, r, l)
-	})
+	femto.OnPost(mux, "/api/admin/add_workstation", authentication.AuthWrapPost(auth, api.addMachine))
```
Comment (aDotInTheVoid) [inline internal/webapi/server.go:60]: NIT Aside: These wrapped names are too long, I want to rename them

File: internal/webapi/server_test.go (hunk 3 above)
Comment (YRMYJASKA) [reply internal/webapi/server_test.go:75]: I test that directly below that

Comment (aDotInTheVoid) [reply internal/webapi/server_test.go:75]: Oh, ignore me, I can't read

File: internal/webapi/server.go (hunk 1 above)
Comment (aDotInTheVoid) [reply internal/webapi/server.go:138]: Turns out you don't need this!

**Pull request merged**